### PR TITLE
Cameronbaird/update jna

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Update jna to v5.10.0 and javapackages-tools to v6.0.0.

javapackages-tools was reverted previously to 5.3.0 in https://github.com/microsoft/CBL-Mariner/pull/2342. This upgrade includes a patch (undeprecate_add_maven_depmap.patch) which keeps the legacy macros in the codebase. Once we have maven in a usable state, we can switch over to the newer %mvn_* macros. 

Move required dependencies out of SPECS-EXTENDED: byaccj hamcrest java-cup jflex junit objectweb-asm qdox xml-commons-resolver xml-commons-apis antlr xerces-j2 xalan-j2 bcel jlex serveletapi5

###### Change Log  

- Update jna to v5.10.0
- Update javapackages-tools to v6.0.0
- Move required dependencies out of SPECS-EXTENDED: byaccj hamcrest java-cup jflex junit objectweb-asm qdox xml-commons-resolver xml-commons-apis antlr xerces-j2 xalan-j2 bcel jlex serveletapi5

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Pipeline build id: in progress...
- local mock build

### CODE REVIEW

File: SPECS/jna/ant_tools_arbitrary_ow2.patch [hunk 1]
```diff
@@ -0,0 +1,66 @@
+diff --git a/build-ant-tools.xml b/build-ant-tools.xml
+index bb73fc74..b0e93f3d 100644
+--- a/build-ant-tools.xml
++++ b/build-ant-tools.xml
+@@ -4,7 +4,7 @@
+     <property name="build" value="build" />
+     <target name="default">
+         <mkdir dir="${build}/ant-tools" />
+-        <javac classpath="lib/ant.jar:lib/asm-8.0.1.jar"
++        <javac classpath="lib/ant.jar:lib/objectweb-asm_asm.jar"
```
Comment (Camelron) [inline SPECS/jna/ant_tools_arbitrary_ow2.patch:10]: Callout: currently we build objectweb 7.2 and coerce it into this spot where 8.0.1 is expected... objectweb-asm is not a trivial upgrade as the Sources schema was complex and they seem to have been moved somewhere else.

File: SPECS/jna/jna.spec [hunk 2]
```diff
@@ -70,12 +87,17 @@ rm -rf dist
 dos2unix OTHERS
 
 %patch0 -p1
+%patch1 -p1
 
+chmod -Rf a+rX,u+w,g-w,o-w .
 sed -i 's|@LIBDIR@|%{_libdir}/%{name}|' src/com/sun/jna/Native.java
 
+build-jar-repository -s -p lib junit ant objectweb-asm/asm hamcrest/core
+rm test/com/sun/jna/StructureFieldOrderInspector.java
```
Comment (Camelron) [inline SPECS/jna/jna.spec:96]: Requires reflections

**Review by mfrw — COMMENTED** [review]
I modified the title of the PR. Please feel free to change to something more informative. 
I think we need to have a consistent way of `commit` messages and `PR` titles :)

A few opinionated resources:
- [Linux kernel commit best practices](https://www.kernel.org/doc/html/v4.12/process/submitting-patches.html) [I personally follow it]
- [How to write a good commit message](https://cbea.ms/git-commit/)

tl;dr
A simple rule that removes a lot of confusion:

"If applied this commit will: `subsystem/spec/package: subject of my commit message`"

PS: None of the above is the `*right*` way; its just, among others, one of the ways :)

File: SPECS/xml-commons-apis/xml-commons-apis.spec [hunk 3]
```diff
@@ -37,7 +37,10 @@ Source4:        https://repo1.maven.org/maven2/xml-apis/xml-apis-ext/1.3.04/xml-
 BuildRequires:  ant
 BuildRequires:  fdupes
 BuildRequires:  java-devel
+BuildRequires:  javapackages-filesystem
```
Comment (PawelWMS) [inline SPECS/xml-commons-apis/xml-commons-apis.spec:40]: Since we already have an [SRPM published for this package](https://packages.microsoft.com/cbl-mariner/2.0/preview/extended/srpms/), we should also bump the `Release` tag.

**Review by PawelWMS — COMMENTED** [review]
One more request for specs migrated from `SPECS-EXTENDED`: we try to keep the ones in `SPECS` "cleaner" and better maintained, since they are considered the core of Mariner specs and will most likely stay with us for a while. Considering that, I'd like to ask you to lint them and remove any macros and comments irrelevant to Mariner, if present. Example: the `%if %{?fedora}` statement will always be false, so we just replace them with the correct branch of the `%if` statement.

File: .github/workflows/validate-cg-manifest.sh [hunk 4]
```diff
@@ -13,6 +13,7 @@
 
 # Ignore some specs, mostly those with Source0 files that are not from an external source, or have very odd URLs
 ignore_list=" \
+  antlr \
```
Comment (jslobodzian) [inline .github/workflows/validate-cg-manifest.sh:16]: Are you sure we need to put these on the ignorelist?  This bypasses registration of these components with legal.  antlr, java-cup, and objectweb-asm don't seem to be ignorable items

Although I think there are a few others on this list too.

**Review by jslobodzian — CHANGES_REQUESTED** [review]

File: .github/workflows/validate-cg-manifest.sh (hunk 4 above)
Comment (Camelron) [reply .github/workflows/validate-cg-manifest.sh:16]: The rationale was more along the lines of not knowing where the source tars come from. Cgmanifest needs a link to where we got the code from, but in cases like java-cup, this code doesn't seem to be hosted anywhere, or at least not in the right version (the tars' hashes don't match).

So a couple of questions about procedure here:
- What Source0 url, if any, should I use if the source tar is manually generated by some generate_tar.sh?
- What Source0 url, if any, should I use if there _is_ a source published online, but not in the version the spec is currently on?
- What Source0 url, if any, should I use if there is no external source to download from whatsoever?

File: SPECS/SymCrypt-OpenSSL/SymCrypt-OpenSSL.spec [hunk 5]
```diff
@@ -13,6 +13,7 @@ BuildRequires:  SymCrypt
 BuildRequires:  cmake
 BuildRequires:  gcc
 BuildRequires:  make
+ExclusiveArch:  x86_64
```
Comment (PawelWMS) [inline SPECS/SymCrypt-OpenSSL/SymCrypt-OpenSSL.spec:16]: This and same change inside `SymCrypt.spec` seem unrelated to the topic. Bad merge? If we do need it, let's also increment the release and add a changelog entry.

File: SPECS/antlr/antlr.spec [hunk 6]
```diff
@@ -1,30 +1,14 @@
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
-#
-# spec file for package antlr
-#
-# Copyright (c) 2020 SUSE LLC
```
Comment (PawelWMS) [inline SPECS/antlr/antlr.spec:6]: Let's keep the copyright notice. If it's not ours, we shouldn't remove it.

File: SPECS/byaccj/byaccj.spec [hunk 7]
```diff
@@ -1,31 +1,13 @@
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
-#
-# spec file for package byaccj
-#
-# Copyright (c) 2014 SUSE LINUX Products GmbH, Nuernberg, Germany.
```
Comment (PawelWMS) [inline SPECS/byaccj/byaccj.spec:6]: Same comment for the copyright notice here and for any other spec, which might need it restored.

File: SPECS/java-cup/java-cup.spec [hunk 8]
```diff
@@ -1,65 +1,48 @@
...
+Source0:        http://www2.cs.tum.edu/projects/cup/releases/%{name}-src-%{cvs_version}-%{pub_date}.tar.gz#/%{name}-%{version}b.tar.gz
 Source1:        java-cup.script
 Source2:        java-cup-generated-files.tar.bz2
 # From          http://www2.cs.tum.edu/projects/cup/
 Source3:        java-cup.license
 Patch1:         java-cup-no-classpath-in-manifest.patch
-Patch2:         java-cup-no-cup-no-jflex.patch
-Patch3:         java-cup-classpath.patch
-# Missing symbolFactory initialization in lr_parser, causes sinjdoc to crash
-Patch4:         java-cup-lr_parser-constructor.patch
```
Comment (PawelWMS) [inline SPECS/java-cup/java-cup.spec:42]: I think we forgot to remove this one and it's still in the `SPECS/java-cup` folder.

File: SPECS/java-cup/java-cup.spec [hunk 9]
```diff
@@ -1,65 +1,48 @@
...
-Summary:        LALR Parser Generator in Java
+Release:        31%{?dist}
 License:        HPND
 Group:          Development/Libraries/Java
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
 Url:            http://www2.cs.tum.edu/projects/cup/
-# https://www2.in.tum.de/WebSVN/dl.php?repname=CUP&path=/develop/&rev=0&isdir=1
-Source0:        develop.tar.bz2
+# http://www2.cs.tum.edu/projects/cup/releases/%{name}-src-%{cvs_version}-%{pub_date}.tar.gz
```
Comment (PawelWMS) [inline SPECS/java-cup/java-cup.spec:16]:
```suggestion
```

File: SPECS/jflex/jflex-bootstrap.spec [hunk 10]
```diff
@@ -149,18 +73,17 @@ mkdir -p %{buildroot}%{_bindir}
 install -p -m 0755 %{SOURCE1} %{buildroot}%{_bindir}/jflex
 
 %files
-%doc COPYRIGHT src/README src/changelog
+%doc jflex/COPYRIGHT jflex/src/README jflex/src/changelog
```
Comment (PawelWMS) [inline SPECS/jflex/jflex-bootstrap.spec:76]: We'd also need something for the `%license` macro.

File: SPECS/jflex/jflex.spec [hunk 11]
```diff
@@ -1,60 +1,31 @@
...
 Group:          Development/Libraries/Java
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
 Url:            http://www.jflex.de/
-Source0:        http://www.jflex.de/jflex-%{version}.tar.bz2
+Source0:        https://github.com/jflex-de/jflex/archive/refs/tags/release_1_4_3.tar.gz#/%{name}-%{version}.tar.gz
 Source1:        jflex.script
 Source100:      jpackage-bootstrap-prepare.sh
 Patch0:         jflex-javac-no-target.patch
 Patch1:         jflex-no-cup-no-jflex.patch
```
Comment (PawelWMS) [inline SPECS/jflex/jflex.spec:14]: Is `Patch1` used in the non-bootstrap version?

File: SPECS/jflex/jflex.spec [hunk 12]
```diff
@@ -92,73 +62,67 @@ Design goals The main design goals of JFlex are:
...
-(cd %{buildroot}%{_javadir} && ln -s jflex.jar JFlex.jar)
+ln -s jflex.jar JFlex.jar
+popd
 
 mkdir -p %{buildroot}%{_bindir}
 install -p -m 0755 %{SOURCE1} %{buildroot}%{_bindir}/jflex
 
 %files
-%doc COPYRIGHT src/README src/changelog
+%doc jflex/COPYRIGHT jflex/src/README jflex/src/changelog
```
Comment (PawelWMS) [inline SPECS/jflex/jflex.spec:112]: Same ask here for `%license`.

File: cgmanifest.json [hunk 13]
```diff
@@ -250,6 +250,16 @@
         }
       }
     },
+    {
+      "component": {
```
Comment (PawelWMS) [inline cgmanifest.json:254]: Can't add a comment in the files itself: please also lint `xz-java.spec`.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
We're close! A few comments about licenses and copyright notices.

File: SPECS/SymCrypt-OpenSSL/SymCrypt-OpenSSL.spec (hunk 5 above)
Comment (Camelron) [reply SPECS/SymCrypt-OpenSSL/SymCrypt-OpenSSL.spec:16]: Huh, I wonder why this ended up as part of the PR when I am based after that PR was merged..

File: SPECS/java-cup/java-cup.spec (hunk 8 above)
Comment (Camelron) [reply SPECS/java-cup/java-cup.spec:42]: good catch!

File: SPECS/antlr/antlr.spec [hunk 14]
```diff
@@ -19,12 +17,15 @@ Distribution:   Mariner
 
 %global debug_package %{nil}
 %bcond_with python2
+
+Summary:        Another Tool for Language Recognition
```
Comment (PawelWMS) [inline SPECS/antlr/antlr.spec:21]: Still needs a license verification, I think.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
I think we're close. What still remains:
1. License verification for moved specs.
2. I'd suggest incrementing the `Release` tag for all specs to publish the updated SRPMs. I believe we're going to clean-up SRPMs before we release these changes but it's better to be safe than sorry.

**Review by PawelWMS — APPROVED** [review]
Approving but please clean-up/lint `ant-junit.spec` before you merge. I think we missed this one.

File: SPECS/hamcrest/hamcrest.spec [hunk 15]
```diff
@@ -65,6 +65,7 @@ mocking libraries and UI validation rules.
 %package core
 Summary:        Core API of hamcrest matcher framework.
 Group:          Development/Libraries/Java
+Provides:       mvn(org.hamcrest:hamcrest-core)
```
Comment (PawelWMS) [inline SPECS/hamcrest/hamcrest.spec:68]: Are you sure this is necessary? Current version of `hamcrest-core` from PMC seems to already be aware of this capability:

```bash
root [ / ]# dnf -q repoquery --whatprovides "mvn(org.hamcrest:hamcrest-core)"
hamcrest-core-0:1.3-14.cm2.noarch
```

File: SPECS/java-cup-nogit.patch.in [hunk 16]
```diff
@@ -0,0 +1,60 @@
+--- a/build.xml
++++ b/build.xml
```
Comment (jslobodzian) [inline SPECS/java-cup-nogit.patch.in:2]: Probably didn't intend to put this file here.

**Review by jslobodzian — APPROVED** [review]
